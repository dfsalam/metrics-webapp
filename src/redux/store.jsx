import { configureStore } from '@reduxjs/toolkit';
import townsReducer from './towns/townsSlice';

const store = configureStore({
  reducer: {
    towns: townsReducer,
  },
});

export default store;
