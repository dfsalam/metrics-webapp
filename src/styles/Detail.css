.detail-container {
  background-color: rgb(225, 84, 150);  
}

.detail-container h2 {
  font-family: 'Gill Sans', serif;
  color: white;
  margin-left: 5px;
  padding-bottom: 0;
  margin-bottom: 5px;
  margin-top: 2px;
}

.detail-container h3 {
  font-family: 'Gill Sans', serif;
  color: white;
  margin-left: 5px;
  padding-bottom: 0;
  margin-bottom: 5px;
  margin-top: 2px;
}