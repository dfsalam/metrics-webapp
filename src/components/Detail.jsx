import React from 'react';
import PropTypes from 'prop-types';
import { useAppSelector } from '../redux/redux-hooks';
import '../styles/Detail.css';
import image from '../img/map.jpg';

const Detail = ({ town }) => {
  const { polution } = useAppSelector((state) => state.polution);
  if (polution.length === 0) {
    return (<h3>Is loading...</h3>);
  }
  const { lat, lon } = polution.coord;
  const {
    co, no, no2, o3, pm25, pm10, so2,
  } = polution.list[0].components;
  return (
    <div className="detail-container">
      <h2>{town}</h2>
      <div>
        <div>
          <h3>
            Latitud:
            {' '}
            {lat}
          </h3>
          <h3>
            Longitud:
            {' '}
            {lon}
          </h3>
        </div>
        <img alt="map" src={image} />
      </div>
      <h3>
        CO (Carbon monoxide),
        {' '}
        {co}
        {' '}
        μg/m3
      </h3>
      <h3>
        NO (Nitrogen monoxide),
        {' '}
        {no}
        {' '}
        μg/m3
      </h3>
      <h3>
        NO2 (Nitrogen dioxide),
        {' '}
        {no2}
        {' '}
        μg/m3
      </h3>
      <h3>
        O3 (Ozone),
        {' '}
        {o3}
        {' '}
        μg/m3
      </h3>
      <h3>
        SO2 (Sulphur dioxide),
        {' '}
        {no}
        {' '}
        μg/m3
      </h3>
      <h3>
        PM2.5 (Fine particles matter),
        {' '}
        {pm25}
        {' '}
        μg/m3
      </h3>
      <h3>
        PM10 (Coarse particulate matter),
        {' '}
        {pm10}
        {' '}
        μg/m3
      </h3>
      <h3>
        NH3 (Ammonia),
        {' '}
        {so2}
        {' '}
        μg/m3
      </h3>
    </div>
  );
};

Detail.propTypes = {
  town: PropTypes.string.isRequired,
};

export default Detail;
