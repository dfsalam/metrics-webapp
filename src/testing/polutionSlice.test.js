import { render, screen } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Detail from '../components/Detail';
import store from '../redux/store';

const mockStore = configureStore([]);

describe('Polution redux state tests', () => {
  it('Expect to initially set polution to an empty object', () => {
    const state = store.getState().polution;
    const initialState = {
      polution: [], isLoading: true,
    };
    expect(state).toEqual(initialState);
  });
  it('Expect to display polution data from mockStore', () => {
    const polution = {
      coord: {
        lon: '-73.3616',
        lat: '5.5324',
      },
      list: [
        {
          main: {
            aqi: 1,
          },
          components: {
            co: '293.73',
            no: '3.69',
            no2: '6.17',
            o3: '0.31',
            so2: '0.33',
            pm2_5: '5.04',
            pm10: '5.69',
            nh3: '1.77',
          },
          dt: '1679033943',
        },
      ],
    };

    const state = {
      polution: {
        polution,
        isLoading: true,
      },
    };
    const store = mockStore(state);
    render(
      <Provider store={store}>
        <Detail town="Tunja" />
      </Provider>,
    );
    const name = screen.getByText('CO (Carbon monoxide), 293.73 Î¼g/m3');
    expect(name).not.toBeNull();
    expect(store.getState).toMatchSnapshot();
  });
});
