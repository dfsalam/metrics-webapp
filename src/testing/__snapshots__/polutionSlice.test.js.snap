// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Polution redux state tests Expect to display polution data from mockStore 1`] = `[Function]`;
